src/manifolds/ProductManifold.jl

```diff
@@ -578,6 +578,24 @@ function sharp!(M::ProductManifold, v::TFVector, x, w::CoTFVector)
     return v
 end
 
+function show(io::IO, M::ProductManifold)
+    length(M.manifolds) == 0 && return println(io, "ProductManifold()")
+    print(io, "ProductManifold(")
+    width = displaysize(io)[2]
+    strings = []
+    one_line_length = 17 + (length(M.manifolds) - 1) * 2
+    for m in M.manifolds
+        s = repr(m)
+        push!(strings, s)
+        one_line_length += length(s)
+    end
+    if one_line_length ≤ width
+        print(io, join(strings, ", "), ")")
```

Comment:
Displaywise an `⨉` might also be nice, but it would not yield constructor with the `PowerManifold` upfront.

Reply:
Yeah, for a small number of manifolds, that would look really nice, but for a large number, not so much. We could do something like 

```julia
Rotations(3)
 × Sphere(3)
 × Euclidean(3; field = ℝ)
 × Stiefel(3, 2, ℝ)
 × Grassmann(3, 2, ℝ)
```

or do dots just showing the first 2 and last 2 submanifolds like

```julia
Rotations(3) × Sphere(3) × … × Stiefel(3, 2, ℝ) × Grassmann(3, 2, ℝ)
```

Either way we should probably do something special to avoid printing all submanifolds for large product manfiolds.

Reply:
For really large ones, I am not so sure, yes. That would be something like:
If this manifold and the last manifold do not fit into this line, but ` ... x` followed by the last manifold into the line. That would yield nearly your second line, with the exception that the ... would be before the last manifold not before the second to last.